3.4.0 damagelogs TTT2 ERROR KILL

Hello, in the new version, when a person is killed, an error pops up, and the corpse disappears.
TTT2 v0.9.2b used.

[tttdamagelogs-3.4.0] addons/tttdamagelogs-3.4.0/lua/damagelogs/shared/events/kills.lua:50: attempt to call method 'HasTeamRole' (a nil value)
  1. unknown - addons/tttdamagelogs-3.4.0/lua/damagelogs/shared/events/kills.lua:50
   2. func - addons/tttdamagelogs-3.4.0/lua/damagelogs/shared/events.lua:59
    3. unknown - lua/includes/modules/hook.lua:198
     4. FireBullets - [C]:-1
      5. ShootBullet - gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:582
       6. unknown - gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:482


[ERROR] Warning! A net message (DL_Ded) is already started! Discarding in favor of the new message! (TTT2RemoveWeaponFromInventory)
  1. RemoveWeaponFromInventoryAndNotifyClient - gamemodes/terrortown/gamemode/server/sv_inventory.lua:36
   2. unknown - gamemodes/terrortown/gamemode/server/sv_weaponry.lua:634
    3. FireBullets - [C]:-1
     4. ShootBullet - gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:582
      5. unknown - gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:482

--- a/lua/damagelogs/server/autoslay.lua
+++ b/lua/damagelogs/server/autoslay.lua
@@ -506,8 +506,12 @@ hook.Add("TTTBeginRound", "Damagelog_AutoSlay", function()
                     ply:SetCleanRound(false)
                     ply:SetNWBool("body_found", true)
 
-                    if ply:GetRole() == ROLE_TRAITOR or TTT2 and ply:HasTeamRole(TEAM_TRAITOR)
-                            or CR_VERSION and ply:IsTraitorTeam() then
+                    print("ply:GetRole() " .. tostring(ply:GetRole()))
+                    print("ply:IsRole(")
+
+                    if ply:GetRole() == ROLE_TRAITOR
+                      or TTT2 and ply:GetTeam() == TEAM_TRAITOR
+                      or CR_VERSION and ply:IsTraitorTeam() then
                         SendConfirmedTraitors(GetInnocentFilter(false))
                     end
 

--- a/lua/damagelogs/shared/events/kills.lua
+++ b/lua/damagelogs/shared/events/kills.lua
@@ -47,8 +47,8 @@ function event:DoPlayerDeath(ply, attacker, dmginfo)
         net.Start("DL_Ded")
 
         if attacker:GetRole() == ROLE_TRAITOR and (ply:GetRole() == ROLE_INNOCENT or ply:GetRole() == ROLE_DETECTIVE)
-                or TTT2 and attacker:HasTeamRole(TEAM_TRAITOR) and not ply:HasTeamRole(TEAM_TRAITOR)
-                or CR_VERSION and attacker:IsTraitorTeam() and not ply:IsTraitorTeam() then
+          or TTT2 and attacker:GetTeam() == TEAM_TRAITOR and not attacker:IsInTeam(ply)
+          or CR_VERSION and attacker:IsTraitorTeam() and not ply:IsTraitorTeam() then
             net.WriteUInt(0, 1)
         else
             net.WriteUInt(1, 1)

